src/node_wasi.cc

```diff
@@ -0,0 +1,87 @@
+#include <cerrno>
+#ifdef _WIN32
+#include <io.h>
+#else
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include "util-inl.h"
+#include "node_internals.h"
+
+#define NAPI_EXPERIMENTAL
+#include "node_api.h"
```

Comment:
Tbh, using N-API in C-only code is going to both be slower *and* less maintainable than using the V8 API directly, at least in my opinion… if we do want to use N-API in our own codebase, we should add the C++ wrapper as a dependency as well

Reply:
i do that because i also have to maintain this as a npm package, and i really don't want to deal with v8 api in the wild.